.github/workflows/build.yml

```diff
@@ -335,29 +354,49 @@ jobs:
 
       - name: Build iOS app
         run: |
-          ./Scripts/build.sh
+          ./Scripts/build.sh ${{ env.BUILD_VARIANT }} "platform=iOS Simulator,name=${{ env.SIMULATOR_NAME || env.DEFAULT_SIMULATOR_NAME }},OS=${{ env.SIMULATOR_VERSION || env.DEFAULT_SIMULATOR_VERSION }}"
```

Comment:
🤔 I feel a little bit confusing to have the simulator destination to be passed on Production build variant, even if that's not used inside `build.sh`. Although I understand is done this way not to duplicate this step adding more conditions to the workflow. However, it's like one needs to enter `build.sh` to understand the second parameter is only used in "Simulator" build variant which is not ideal IMO.

Reply:
I don't love it, either. The other option is to have different build scripts for the two flows—simulator build or non-simulator build—but that could potentially run into the bifurcation problem. Neither solution seems particularly great to me, so I defaulted to the "one build script" side.

I suppose a third option would be to make it an optional parameter of the build, but then we'd have the step duplication.

@vvolkgang do you have an opinion on this?

Reply:
I think I've used `'generic/platform=iOS Simulator'` as the destination for a simulator build before and then you don't have to specify a specific device and version.

Reply:
When I was doing it locally I had to specify the whole thing out, but I don't think I had the "generic" term in there 🤔 I'll double-check and see

Reply:
Oh hey, with the generic in there it works, and I can just pull it into the build script and not have to pass it in anymore at all